### PR TITLE
mkfs: Improve fat length computation

### PR DESCRIPTION
**Pull request by lvxnull**
This accounts for fat offset and may result in smaller fat sizes.

### CODE REVIEW

File: mkfs/mkfs.c [hunk 1]
```diff
@@ -467,14 +469,16 @@ static int exfat_build_mkfs_info(struct exfat_blk_dev *bd,
 	}
 	finfo.fat_byte_off = round_up(bd->offset + 24 * bd->sector_size,
 			ui->boundary_align) - bd->offset;
+
 	/* Prevent integer overflow when computing the FAT length */
-	if (bd->num_clusters > UINT32_MAX / 4) {
+	max_clusters = (bd->size - finfo.fat_byte_off - 8 * num_fats - 1)/(ui->cluster_size + 4 * num_fats) + 1;
```
Comment (namjaejeon) [inline mkfs/mkfs.c:474]: Could you explain this calculation in more detail? And please add the meaning of this calculation in the /* */ comment.

Comment (lvxnull) [reply mkfs/mkfs.c:474]: In the optimal scenario, there are exactly as many entries in the FAT as there are clusters on the volume + 2 unused entries in the beginning. So we have $$l_f = 4(n_c + 2) = \frac{l_V - o_f - n_cl_c}{n_f}$$ where
- $l_f$ - length of fat
- $l_V$ - length of volume
- $l_c$ - length of cluster
- $n_c$ - number of clusters
- $o_f$ - fat offset
Now solving for $n_c$ and implementing ceil division as $\lfloor{\frac{a - 1}{b} + 1}\rfloor$ we get exactly this formula.

Comment (lvxnull) [reply mkfs/mkfs.c:474]: I don't know how to this entire explanation in a C comment, would a link to this comment suffice?

Comment (namjaejeon) [reply mkfs/mkfs.c:474]: Okay, If so, can you update a patch description with this explanation ?

**Comment by namjaejeon** [comment]
Applied it to #exfat-next now.
Thanks for your patch!

**Pull request closed without merging**